tests/unit/test_charm.py

```diff
@@ -308,3 +333,37 @@ def test_pebble_ready(harness: Harness, patch: SpringBootPatch):
     harness.begin_with_initial_hooks()
     harness.container_pebble_ready("spring-boot-app")
     assert isinstance(harness.model.unit.status, ActiveStatus)
+
+
+def test_datasource(harness: Harness, patch: SpringBootPatch):
+    """
+    arrange: provide a simulated Spring Boot application image.
+    act: start the charm and a database. Integrates the db to the charm.
+    assert: test datasource output of the charm when integrated with a db or not.
```

Comment:
I'd specify what is expected for that datasource output? e.g. datasource contains the data needed to establish a connect

Reply:
Improved